src/js/tech/tech.js

```diff
@@ -813,6 +813,8 @@ Tech.withSourceHandlers = function(_Tech){
     // than clear all of our current tracks
     if (this.currentSource_) {
       this.clearTracks(['audio', 'video']);
```

Comment:
Ended up removing clearing out the `currentSource_` in `disposeSourceHandler` and only doing it inside this if statement.